### PR TITLE
[TableGen] Enhance testability of TableGen-based macro fusion

### PR DESCRIPTION
**Pull request by wangpc-pp**
We add an option `-mfusion` like `-mattr` to add/remove supported
macro fusions, so that we can test each macro fusion separately via
`llc`.

This PR is stacked on #72219, #72222, #72223

### CODE REVIEW

File: llvm/lib/CodeGen/TargetSubtargetInfo.cpp [hunk 1]
```diff
@@ -14,16 +14,62 @@
...
+      char Prefix = MFusion[0];
+      bool Disable = Prefix == '-';
+      if (Prefix == '+' || Prefix == '-')
+        MFusion = MFusion.substr(1);
+
+      // MacroFusionTable is sorted.
+      const auto *Pos = std::lower_bound(
+          MacroFusionTable.begin(), MacroFusionTable.end(), MFusion,
+          [](const MacroFusionEntry &LHS, const std::string &RHS) {
+            int CmpName = StringRef(LHS.Name).compare(RHS);
```
Comment (topperc) [inline llvm/lib/CodeGen/TargetSubtargetInfo.cpp:48]: Isn't this just `return String(LHS.Name).compare(RHS) < 0`

**Comment by topperc** [comment]
Could we have tblgen autogenerate the macrofusion bits onto the end of the existing subtarget feature bits and autogenerate `has*()` functions for them like other subtarget features. It seems like we're now duplicating some of the -mattr functionality.

**Comment by wangpc-pp** [comment]
> Could we have tblgen autogenerate the macrofusion bits onto the end of the existing subtarget feature bits and autogenerate `has*()` functions for them like other subtarget features. It seems like we're now duplicating some of the -mattr functionality.

Yes, I was thinking about making `Fusion` inherit from `SubtargetFuture` so that we can reuse current `-mattr` mechanism. My concern is that `MAX_SUBTARGET_FEATURES` is 320 now and there are more 170 subtarget features in RISCV (270 for AArch64, 223 for ARM and 183 for X86). For the foreseeable future, RISCV, as a modular ISA, may have more than 320 features. We may need to increase `MAX_SUBTARGET_FEATURES` in the near future if we add macrofusion bits to subtarget feature bits.
But it may not be a big problem, I will post another patch for this solution.

**Comment by wangpc-pp** [comment]
Close this PR as I prefer #73115.

**Pull request closed without merging**